### PR TITLE
Add Collector Init() interface

### PR DESCRIPTION
**Pull request by robingustafsson**
Base changes for implementing https://github.com/loadimpact/k6/issues/156. I'll amend @martinfijal WIP commit on `feature/cloud` branch with changes to cloud collector.

### CODE REVIEW

File: lib/collector.go [hunk 1]
```diff
@@ -22,11 +22,16 @@ package lib
 
 import (
 	"context"
+
```
Comment (uppfinnarn) [inline lib/collector.go]: I'm just gonna run `goimports` on everything and merge #121

**Comment by uppfinnarn** [comment]
Hm, my idea was to call `Init()` right after printing the Web UI URL and let collectors print their own "result url: [URL]", but maybe it looks better for cloud results to just have "output: [URL]" instead?

File: lib/collector.go (hunk 1 above)
Comment (robingustafsson) [inline lib/collector.go]: 👍 Do it!

**Comment by robingustafsson** [comment]
Not sure what I think looks best. Yeah, maybe just have the URL as the value for `output`, now it's in parentheses "Load Impact (URL)". Or would that be weird if for example the influxdb collector would point to a `localhost` address, I mean would it be non-obvious that the `influxdb` collector is being used?

Let me know what you prefer and I'll fix the code.

Also, is the circleci lint failure `run.go:438:15:warning: the channel used with signal.Notify should be buffered (SA1017) (staticcheck)` something known?

File: lib/collector.go (hunk 1 above)
Comment (uppfinnarn) [inline lib/collector.go]: Merged!

**Comment by uppfinnarn** [comment]
I'm honestly more worried about the width of "Load Impact (URL)"… how long are URLs typically?

Also, have you considered: `output: load impact / http://...`

**Comment by robingustafsson** [comment]
The base URL is 43 chars with the staging URL, will be 35 for prod, and then some chars for the numeric test run ID. So plenty of space if we're aiming for 80 chars :-)

**Comment by uppfinnarn** [comment]
The linter should be fine if you merge `master` now, let's go with this for now.

**Comment by robingustafsson** [comment]
Now rebased on `master`.

**Comment by robingustafsson** [comment]
@uppfinnarn We got a `FAIL: TestGetGroups/list/groups` in CI, doesn't seem related to my changes, something known?

**Comment by uppfinnarn** [comment]
Urgh, that's #76, and I have no idea why this happens or what to do about it ><

Just, rebuild and it usually works next time

**Comment by robingustafsson** [comment]
@uppfinnarn Ah, thanks. Now rebuilt and passing!

**Pull request merged**